src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -759,8 +759,17 @@ private void Parse()
                 @"((?<username>[^:@]+)(:(?<password>[^:@]*))?@)?" +
                 serversPattern + @"(/" + databasePattern + ")?/?" + optionsPattern + "$";
 
+            Match badPercentMatch = null;
+            var hexDigit= "0-9A-Fa-f";
+            var percentPattern = $"(%[^{hexDigit}])|"  // % followed by a non-hexadecimal digit OR
+                                 +$"(%[{hexDigit}][^{hexDigit}])"; // % followed by a hexadecimal digit and a non-hexadecimal digit
```

Comment:
Ah, sorry! We tend to use a four space-indent for multiline statements (and one space before and after operators like `+`)
The dev guide was unclear about this, and I've updated it accordingly